fish-rust/src/threads.rs

```diff
@@ -49,6 +49,21 @@ static NOTIFY_SIGNALLER: once_cell::sync::Lazy<&'static crate::fd_monitor::FdEve
         result
     });
 
+/// A tuple of `(ThreadId, JoinHandle)` of all threads we've (directly) spawned that haven't yet
```

Comment:
This contains a LOT of code to work around an ASAN limitation. I'd be OK with a line or two, but don't think we should do something so invasive. I'd rather just disable ASAN's leak detection than complicate our own code in this way.